### ISSUE TITLE
4.9.4 starts reporting does not satisfy the constraint, on non active extends cases

### ISSUE THREAD

**Issue by rostislav-simonik** [microsoft/TypeScript#51928]
# Bug Report

### 🔎 Search Terms

does not satisfy the constraint

### 🕗 Version & Regression Information
4.9.4

- This changed between versions 4.8.4 and 4.9.4

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=4.9.4#code/JYOwLgpgTgZghgYwgAgPICMBWEFgM7IDeAUMsgEJxQBcyA5JVAIx0DcpyAYgPbe109uLdgF9i4iAA8ADtyhhkYAJ7SUGbLgBycALYQCAXmQBrCEu4w0WHPnbjlq5AHEIYAGpwANgFcIAFRUIAB5tPWQpSBAAEwJ1G1D9AD5kIzjcPABtBIBde0DkAEEU+gBlbj06cMlImKsNMAS8DjIAfmdXDx9-QKC6MorE5uR+ABEIeG9PMDoOcQ4gA

### 💻 Code

```ts
interface Objects {
  Bar: 'Bar1';
  Foo: 'Foo1';
}


export type ObjectNames = keyof Objects;


type GetValueType<Name extends ObjectNames> = Objects[Name]

type A = 'Some' extends ObjectNames
    ? GetValueType<'Some'> /// <<< here is the issue
    : 'Default'
```

### 🙁 Actual behavior

`Type '"Some"' does not satisfy the constraint 'keyof Objects'`

### 🙂 Expected behavior

Shouldn't report, because given case doesn't meet extends criteria.

**Comment by RyanCavanaugh** [comment]
This is the correct behavior; 4.8 and before had a bug. You can write
```ts
type A = 'Some' extends ObjectNames ? GetValueType<'Some' & ObjectNames> : 'Default'
```

**Comment by rostislav-simonik** [comment]
Thanks for the quick response.
